omnibus/cookbooks/omnibus-supermarket/libraries/config.rb

```diff
@@ -58,21 +58,71 @@ def self.load_or_create_secrets!(filename, node)
       node.consume_attributes('supermarket' => secrets)
     end
 
-    def self.audit_config(node)
+    def self.audit_config(config)
+      audit_s3_config(config)
+      audit_fips_config(config)
+    end
+
+    def self.audit_s3_config(config)
       required_s3_vars = %w(s3_bucket s3_access_key_id s3_secret_access_key s3_region).freeze
-      any_s3_settings = required_s3_vars.any? { |key| !(node[key].nil? || node[key].empty?) }
-      all_s3_settings = required_s3_vars.all? { |key| !(node[key].nil? || node[key].empty?) }
+      any_s3_settings = required_s3_vars.any? { |key| !(config[key].nil? || config[key].empty?) }
+      all_s3_settings = required_s3_vars.all? { |key| !(config[key].nil? || config[key].empty?) }
 
       if any_s3_settings && !all_s3_settings
         raise IncompleteConfig.new "Got some, but not all, of the required S3 configs. Must provide none or all of #{required_s3_vars} to configure cookbook storage in an S3 bucket."
       end
 
-      if node['s3_bucket'] =~ /\./ &&
-         (node['s3_domain_style'] != ':s3_path_url' || node['s3_region'] != 'us-east-1')
+      if config['s3_bucket'] =~ /\./ &&
+         (config['s3_domain_style'] != ':s3_path_url' || config['s3_region'] != 'us-east-1')
          raise IncompatibleConfig.new "Incompatible S3 bucket settings. If the bucket name contains periods, the bucket must be in us-east-1 and the domain style must be :s3_path_url.\nAmazon recommends against periods in bucket names. See: https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html"
       end
     end
 
+    def self.audit_fips_config(config)
+      if !built_with_fips?(config['install_directory'])
+        if fips_enabled_in_kernel?
+          raise IncompatibleConfig.new "Detected FIPS is enabled in the kernel, but FIPS is not supported by this installer."
+        end
+        if config['fips_enabled']
```

Comment:
These checks imply that it's an error when fips is supported in kernel, but is not enabled in config?  Or can it run in non-fips mode on a FIPs kernel? (If so, the enable check + non-fips build may be incorrect.)